### PR TITLE
Added support for non-8MHz master clock

### PR DESCRIPTION
**Pull request by OPLx**
Hello,

This is what I had modified a while back to support different master clocks.  I tried to keep the style and format the same as the original source.  It's a little different from your newer version, so I'm not sure if would be the ideal pull request candidate.  I thought I'd post it here for you to peruse.

Please let me know what you think whenever you have time.

### CODE REVIEW

File: src/SAAImpl.cpp [hunk 1]
```diff
@@ -82,11 +85,11 @@ m_uParamRate(0)
 
 	// Create amplification/mixing stages and link to appropriate oscillators,
 	// noise generators and envelope controlloers
-	Amp[0] = new CSAAAmp(Osc[0], Noise[0], NULL),
-	Amp[1] = new CSAAAmp(Osc[1], Noise[0], NULL),
-	Amp[2] = new CSAAAmp(Osc[2], Noise[0], Env[0]),
-	Amp[3] = new CSAAAmp(Osc[3], Noise[1], NULL),
-	Amp[4] = new CSAAAmp(Osc[4], Noise[1], NULL),
+    Amp[0] = new CSAAAmp(Osc[0], Noise[0], NULL);
```
Comment (simonowen) [inline src/SAAImpl.cpp]: unnecessary whitespace change?

File: include/SAASound.h [hunk 2]
```diff
@@ -107,7 +110,7 @@ extern "C" {
 typedef void * SAASND;
 
 // "C-style" interface for the CSAASound class
-SAASND SAAAPI newSAASND(void);
+SAASND SAAAPI newSAASND(unsigned long nMasterClock);
```
Comment (simonowen) [inline include/SAASound.h]: This breaks compatibility with C applications using the old API.  It would be better to keep the old version with the original clock and a second function with the new clock parameter.

**Comment by simonowen** [comment]
I've added a couple of comments.  If the C API issue is resolved I'll merge it in  :)

File: include/SAASound.h (hunk 2 above)
Comment (OPLx) [reply include/SAASound.h]: Thank you for pointing this out!  I'll go ahead and change that back and introduce a new function instead.

File: src/SAAFreq.cpp [hunk 3]
```diff
@@ -154,6 +160,36 @@ unsigned short CSAAFreq::Level(void) const
...
+{
+    if (nMasterClock > 0)
+    {
+        unsigned short const nMaxOffset = 256;
+        unsigned char const nMaxOctave = 8;
+
+        // static_assert to ensure that octave and offset sizes match that of m_FreqTable
+        SAAFREQ_STATIC_ASSERT((sizeof(m_FreqTable) / sizeof(m_FreqTable[0])) == (nMaxOffset * nMaxOctave));
+
+        unsigned long const freqMulti = static_cast<unsigned long>((nMasterClock / 512.0) + 0.5);
```
Comment (OPLx) [inline src/SAAFreq.cpp]: I just realized that I had used doubles to generate the table; it should (eventually) be using fixed-point calculations instead.  I'll try to change to that hopefully in the near future.

File: src/SAAImpl.cpp (hunk 1 above)
Comment (simonowen) [reply src/SAAImpl.cpp]: ah, I hadn't noticed the difference in commas!  The indentation looked a bit different, so it might be worth matching the surrounding code.

**Comment by simonowen** [comment]
Thanks for the updates!

**Comment by OPLx** [comment]
Thank you!
